### DISCUSSION TITLE
Magit is not compatible with the latest compat.el

### DISCUSSION THREAD

**Discussion by Eason0210** [magit/magit#4737] [Support] [answered]
Hi @tarsius ,

With latest commit of magit [(124a5ad8)](https://github.com/magit/magit/commit/124a5ad8) and compat.el [(99e7939)](https://git.sr.ht/~pkal/compat/commit/99e7939), when use `M-x magit-status`,it will show these warning errors:

```
⛔ Warning (comp): /Users/eason/.emacs.d/lib/magit/lisp/magit-git.el: Error: File is missing Renaming
⛔ Warning (comp): /Users/eason/.emacs.d/lib/magit/lisp/magit-process.el: Error: File is missing Renaming
⛔ Warning (comp): /Users/eason/.emacs.d/lib/magit/lisp/magit-transient.el: Error: File is missing Renaming
⛔ Warning (comp): /Users/eason/.emacs.d/lib/magit/lisp/magit-margin.el: Error: File is missing Renaming
⛔ Warning (comp): /Users/eason/.emacs.d/lib/magit/lisp/magit-refs.el: Error: File is missing Renaming
⛔ Warning (comp): /Users/eason/.emacs.d/lib/magit/lisp/magit-files.el: Error: File is missing Renaming
⛔ Warning (comp): /Users/eason/.emacs.d/lib/magit/lisp/magit-apply.el: Error: File is missing Renaming
⛔ Warning (comp): /Users/eason/.emacs.d/lib/magit/lisp/magit-branch.el: Error: File is missing Renaming
⛔ Warning (comp): /Users/eason/.emacs.d/lib/magit/lisp/magit-reset.el: Error: File is missing Renaming
⛔ Warning (comp): /Users/eason/.emacs.d/lib/magit/lisp/magit-blame.el: Error: File is missing Renaming

```
But it works well if I roll back compat.el to commit [8c51417](https://git.sr.ht/~pkal/compat/commit/8c51417)

And it also works well with [compat 28.1.1.3]( https://git.sr.ht/~pkal/compat/commit/2c3233c) 
But not works with[ compat 28.1.2.0](https://git.sr.ht/~pkal/compat/commit/6b70229)

steps to reproduce the issue:

1. Install Borg(Bootstrapping using a seed)
```bash
git clone git@github.com:emacscollective/emacs.g.git ~/.emacs.d
cd ~/.emacs.d
make bootstrap-borg
make bootstrap
```
2. Start Emacs

3. C-x, C-f ~/.emacs.d/init.el
4. `M-x, magit-status`
5. Update the magit and compat to latest version
6. Restart Emacs
7. C-x, C-f ~/.emacs.d/init.el
8. `M-x, magit-status`, Now you will see the warnings

OS: macOS Monterey 12.5
Emacs 29.0.50 --with-native-compilation

**Comment by tarsius** [comment, marked as answer]
Try running `make clean-force build` after (5).

**Reply by Eason0210** [reply]
I try `make clean-force build`, but the issue still exist.

I also try `make clean` and then `rm -rf eln-cache`,
then start Emacs, and try `magit-status`, looks better, only got one warning
```
⛔ Warning (comp): /Users/eason/.emacs.d/lib/magit/lisp/magit-margin.el: Error: File is missing Renaming
```
restart Emacs and try `magit-status` again, no warning, but got this info in `*Async-native-compile-log*`

```
Compilation finished.
Compiling /Users/eason/.emacs.d/lib/magit/lisp/magit-margin.el...
Compilation finished.
```
Then I restart Emacs again, everything is working well, no native compile and warning info when I use `magit-status`.
